csfieldguide/static/interactives/pixel-viewer/js/pixel-viewer.js

```diff
@@ -996,14 +1017,39 @@ var paint = function(row, col, left, top, width, height, zoom) {
             context.fillStyle = "rgba(110, 110, 110, " + text_opacity + ")";
         }
 
-        // Pretty primitive text positioning :)
-        var cell_lines = cell_text.split('\n');
-
-        for (var i = 0; i < cell_lines.length; i++)
-            context.fillText(cell_lines[i] + pixelData[i], left + (6 * zoom), top + (14 * zoom) + (i * cell_line_height * zoom) );
+        // Pretty primitive text positioning
+        if (colour_code_rep == 'hex') { // Shows colour codes in #FFFFFF style
+          context.font = (10 * zoom).toFixed(2) + 'px Consolas, Courier New, monospace';
+          r = pixelData[0];
+          g = pixelData[1];
+          b = pixelData[2];
+          hex_string = rgbToHex(r, g, b);
+          context.fillText(hex_string, left + (4 * zoom), top + (14 * zoom) + (cell_line_height * zoom));
+        } else {
+          context.font = (14 * zoom).toFixed(2) + 'px Consolas, Courier New, monospace';
+          var cell_lines = cell_text.split('\n');
+          for (var i = 0; i < cell_lines.length; i++) {
+            if (colour_code_rep == 'rgb-hex') { // Shows colour codes in RGB using Hexadecimal
+              value = ' ' + componentToHex(pixelData[i]) // Prefix with space so that the positioning is nice
+            } else { // Shows colour codes in RGB using Decimal
+              value = pixelData[i]
+            }
+            context.fillText(cell_lines[i] + value, left + (6 * zoom), top + (14 * zoom) + (i * cell_line_height * zoom));
+          }
+        }
     }
 };
 
+// Taken from https://stackoverflow.com/questions/5623838/rgb-to-hex-and-hex-to-rgb
+function componentToHex(c) {
+  var hex = c.toString(16);
+  return hex.length == 1 ? "0" + hex : hex;
+}
```

Comment:
I think the hex should be uppercase to have consistent-height digits, but it might get confusing if B (blue) is misinterpreted as B (11)